micrometer-core/src/main/java/io/micrometer/core/instrument/binder/jetty/JettyConnectionMetrics.java

```diff
@@ -160,10 +160,12 @@ private static Tags getConnectorNameTag(Connector connector) {
     @Override
     public void onOpened(Connection connection) {
         Timer.Sample started = Timer.start(registry);
+        int connections;
         synchronized (connectionSamplesLock) {
             connectionSamples.put(connection, started);
-            maxConnections.record(connectionSamples.size());
+            connections = connectionSamples.size();
         }
+        maxConnections.record(connections);
```

Comment:
I assume this was intentional before so that `put` and `record` are atomic. Reading the two values are happening separately though (separate gauges) so we can still end up with ~"inconsistent values".

I don't think this is a big problem though but I think if we don't want to ensure that that `put` and `record` are atomic, maybe we can use a `ConcurrentMap` instead of a lock (should be better because of lock striping).

Reply:
> Reading the two values are happening separately though

That's what led me to believe there's no need to make the 2 updates in a single transaction. The `maxConnections` readingsl\ still capture the correct maximum number of simultaneous connections observed over time (correct in the sense that it's a value `connectionSamples.size()` call would produce at a point in time).

We need an operation that both updates the map and returns its size transactionally here - that is not something `ConcurrentMap` offers